openerp/addons/base/res/res_users.py

```diff
@@ -163,13 +163,10 @@ def _is_share(self, cr, uid, ids, name, args, context=None):
         return res
 
     def _get_users_from_group(self, cr, uid, ids, context=None):
-        result = set()
-        groups = self.pool['res.groups'].browse(cr, uid, ids, context=context)
-        # Clear cache to avoid perf degradation on databases with thousands of users
-        groups.invalidate_cache()
-        for group in groups:
-            result.update(user.id for user in group.users)
-        return list(result)
+        group_user = self.pool['ir.model.data'].xmlid_to_object(cr, SUPERUSER_ID, 'base.group_user', context=context)
+        if group_user and group_user.id in ids:
+            return group_user.users.ids
+        return []
```

Comment:
Changing the semantics of a generic _get_users_from_group() to introduce this strange behavior is extremely odd. I'd rather add a new _get_internal_users_from_group() method, and keep the old method for backwards-compatibility.